aptos-move/move-examples/token_objects/sources/collection.move

```diff
@@ -242,97 +220,71 @@ module token_objects::collection {
 
+    public fun generate_mutator_ref(ref: &ConstructorRef): MutatorRef {
+        let object = object::object_from_constructor_ref<Collection>(ref);
+        MutatorRef { self: object::object_address(&object) }
+    }
+
     // Accessors
-    inline fun verify<T: key>(collection: &Object<T>): address {
+    inline fun borrow<T: key>(collection: &Object<T>): &Collection {
         let collection_address = object::object_address(collection);
         assert!(
             exists<Collection>(collection_address),
             error::not_found(ECOLLECTION_DOES_NOT_EXIST),
         );
-        collection_address
+        borrow_global<Collection>(collection_address)
     }
 
     public fun creator<T: key>(collection: Object<T>): address acquires Collection {
-        let collection_address = verify(&collection);
-        borrow_global<Collection>(collection_address).creator
+        borrow(&collection).creator
     }
 
     public fun description<T: key>(collection: Object<T>): String acquires Collection {
-        let collection_address = verify(&collection);
-        borrow_global<Collection>(collection_address).description
-    }
-
-    public fun is_description_mutable<T: key>(collection: Object<T>): bool acquires Collection {
-        let collection_address = verify(&collection);
-        borrow_global<Collection>(collection_address).mutability_config.description
-    }
-
-    public fun is_uri_mutable<T: key>(collection: Object<T>): bool acquires Collection {
-        let collection_address = verify(&collection);
-        borrow_global<Collection>(collection_address).mutability_config.uri
+        borrow(&collection).description
     }
 
     public fun name<T: key>(collection: Object<T>): String acquires Collection {
-        let collection_address = verify(&collection);
-        borrow_global<Collection>(collection_address).name
+        borrow(&collection).name
     }
 
     public fun uri<T: key>(collection: Object<T>): String acquires Collection {
-        let collection_address = verify(&collection);
-        borrow_global<Collection>(collection_address).uri
+        borrow(&collection).uri
     }
 
     // Mutators
 
-    public fun set_description<T: key>(
-        creator: &signer,
-        collection: Object<T>,
-        description: String,
-    ) acquires Collection {
-        let collection_address = verify(&collection);
-        let collection = borrow_global_mut<Collection>(collection_address);
-        assert!(
-            collection.creator == signer::address_of(creator),
-            error::permission_denied(ENOT_CREATOR),
-        );
-
+    inline fun borrow_mut(mutator_ref: &MutatorRef): &mut Collection {
         assert!(
-            collection.mutability_config.description,
-            error::permission_denied(EFIELD_NOT_MUTABLE),
+            exists<Collection>(mutator_ref.self),
+            error::not_found(ECOLLECTION_DOES_NOT_EXIST),
         );
+        borrow_global_mut<Collection>(mutator_ref.self)
+    }
```

Comment:
Should this be a function in the object standard?

Reply:
there is one called `address_to_object<T>` which I used in all the verification places.